lib/setupmanagers/hckclient.rb

```diff
@@ -17,17 +17,14 @@ class HCKClient
     attr_reader :name, :kit
     attr_writer :support
 
-    def initialize(project, setup_manager, studio, name)
+    def initialize(project, setup_manager, studio, name, run_opts)
       @project = project
       @logger = project.logger
       @studio = studio
       @name = name
       @kit = setup_manager.kit
       @setup_manager = setup_manager
       @pool = 'Default Pool'
-    end
-
-    def run(run_opts = nil)
       @setup_manager.run_client(@name, run_opts)
```

Comment:
As these methods will be more complicated, I suggest adding several debug logs to know what we exactly do now.
And in hckstudio.

Reply:
Done with commit 9bbd4ae218bf81aeae32584f1b00da9ace886ffe.